advanced_filters/views.py

```diff
@@ -69,8 +69,8 @@ def get(self, request, model=None, field_name=None):
                 logger.debug('No choices calculated for field %s of type %s',
                              field, type(field))
             else:
-                choices = set(model_obj.objects.values_list(field.name, flat=True))
-                if len(choices) <= max_choices:
+                choices = model_obj.objects.values_list(field.name, flat=True).distinct()
```

Comment:
Hope you don't mind me commenting while you work, I think `order_by` explicitly here is really a must, since the model may have default ordering that differs from our `field.name`, which [may break the `distinct` query, as noted here][1].

[1]: https://docs.djangoproject.com/en/1.10/ref/models/querysets/#django.db.models.query.QuerySet.distinct